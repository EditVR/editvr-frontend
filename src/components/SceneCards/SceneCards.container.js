/**
 * @file SceneCards.container.js
 * Exports a redux-connected SceneCards component.
 */

import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import SceneCards from './SceneCards';

const mapDispatchToProps = dispatch => ({
  dispatch
});

const mapState = ({ openExperience }) => ({
  experience: openExperience
});

export default withRouter(connect(mapState, mapDispatchToProps)(SceneCards));
