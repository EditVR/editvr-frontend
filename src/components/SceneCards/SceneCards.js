/**
 * @file SceneCards.js
 * Exports a component that renders cards for all of the scenes included in the
 * experience that's currently open.
 */

import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Link } from 'react-router-dom';
import {
  withStyles,
  Typography,
  Card,
  CardContent,
  CardActions,
  CardMedia,
  Button,
  Tooltip
} from '@material-ui/core';
import { OpenInBrowser } from '@material-ui/icons';

import SceneCardsStyles from './SceneCards.style';

const SceneCards = ({
  experience: { item: experience },
  classes,
  match: {
    params: { sceneSlug }
  }
}) => (
  <div>
    {experience.field_scenes &&
      experience.field_scenes.map(
        ({ id, title, field_slug: slug, field_photosphere: photosphere }) => (
          <Card
            key={id}
            raised={sceneSlug === slug}
            className={classNames(classes.card, {
              [classes.cardActive]: sceneSlug === slug
            })}
          >
            {photosphere &&
              photosphere.meta.derivatives && (
                <CardMedia
                  className={classes.cardMedia}
                  title={title}
                  image={photosphere.meta.derivatives.sc}
                />
              )}
            <CardContent>
              <Typography gutterBottom variant="headline">
                {title}
              </Typography>
            </CardContent>
            <CardActions>
              <Tooltip title={`Open ${title}`}>
                <Button
                  variant="outlined"
                  size="small"
                  component={Link}
                  disabled={sceneSlug === slug}
                  className={classes.cardActionButton}
                  to={`/experience/vreditor/${
                    experience.field_experience_path
                  }/${slug}`}
                >
                  <OpenInBrowser />
                </Button>
              </Tooltip>
            </CardActions>
          </Card>
        )
      )}
  </div>
);

SceneCards.propTypes = {
  classes: PropTypes.shape({}).isRequired,
  experience: PropTypes.shape({
    item: PropTypes.shape({
      field_scenes: PropTypes.arrayOf(
        PropTypes.shape({
          id: PropTypes.string,
          title: PropTypes.string,
          field_slug: PropTypes.string,
          field_photosphere: PropTypes.shape({
            meta: PropTypes.shape({
              derivatives: PropTypes.shape({
                sc: PropTypes.string
              })
            })
          })
        })
      )
    })
  }).isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      experienceSlug: PropTypes.string.isRequired,
      sceneSlug: PropTypes.string
    }).isRequired
  }).isRequired
};

export default withStyles(SceneCardsStyles)(SceneCards);
