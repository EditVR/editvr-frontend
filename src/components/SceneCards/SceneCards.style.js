/**
 * @file SceneCards.style.js
 * Exports SceneCards component styles.
 */

export default theme => ({
  card: {
    marginTop: theme.spacing.unit * 2
  },
  cardActive: {
    border: '1px solid',
    borderColor: theme.palette.secondary.dark
  },
  cardActionButton: {
    marginRight: theme.spacing.unit
  },
  cardMedia: {
    height: 0,
    // 16/9 ratio.
    paddingTop: '56.25%'
  }
});
