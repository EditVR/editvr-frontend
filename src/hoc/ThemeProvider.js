/**
 * @file ThemeProvider.js
 * Exports a HOC ThemeProvider component that attaches children their MUI theme.
 */

import React from 'react';
import { MuiThemeProvider } from '@material-ui/core';

import muiTheme from '../styles/muiTheme';

const ThemeProvider = WrappedComponent => props => (
  <MuiThemeProvider theme={muiTheme}>
    <WrappedComponent {...props} />
  </MuiThemeProvider>
);

export default ThemeProvider;
