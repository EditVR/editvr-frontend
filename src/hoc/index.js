/**
 * @file index.js
 * Exports all hoc components.
 */

import ThemeProvider from './ThemeProvider';

export { ThemeProvider };
