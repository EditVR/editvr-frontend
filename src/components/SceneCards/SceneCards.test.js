/**
 * @file SceneCards.test.js
 * Contains tests for SceneCards.js.
 */

import React from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { MemoryRouter as Router, Route } from 'react-router-dom';
import { createMount } from '@material-ui/core/test-utils';

import { ThemeProvider } from '../../hoc';
import SceneCards from './SceneCards.container';

const ThemedSceneCards = ThemeProvider(SceneCards);

describe('<SceneCards />', () => {
  it('Matches its snapshot', () => {
    const store = configureStore()({
      openExperience: {
        item: {
          title: 'test',
          field_experience_path: 'test',
          scenes: {
            test: {
              id: '10',
              title: 'test',
              field_slug: 'test',
              field_photosphere: {
                meta: {
                  derivatives: {
                    sc: 'https://image.path'
                  }
                }
              }
            }
          }
        }
      }
    });

    expect(
      createMount()(
        <Provider store={store}>
          <Router initialEntries={['/experience/vreditor/test/test']}>
            <Route
              path="/experience/vreditor/:experienceSlug/:sceneSlug?"
              component={ThemedSceneCards}
            />
          </Router>
        </Provider>
      ).html()
    ).toMatchSnapshot();
  });
});
