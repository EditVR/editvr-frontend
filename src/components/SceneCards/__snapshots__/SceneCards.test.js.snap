// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SceneCards /> Matches its snapshot 1`] = `"<div><div class=\\"MuiPaper-root-6 MuiPaper-elevation8-16 MuiPaper-rounded-7 MuiCard-root-5 SceneCards-card-1 SceneCards-cardActive-2\\"><div class=\\"MuiCardMedia-root-33 SceneCards-cardMedia-4\\" style=\\"background-image: url(https://image.path);\\" title=\\"test\\"></div><div class=\\"MuiCardContent-root-35\\"><h1 class=\\"MuiTypography-root-36 MuiTypography-headline-41 MuiTypography-gutterBottom-53\\">test</h1><p class=\\"MuiTypography-root-36 MuiTypography-body1-45\\">This scene does not yet have a description.</p></div><div class=\\"MuiCardActions-root-60\\"><a tabindex=\\"0\\" class=\\"MuiButtonBase-root-86 MuiButton-root-70 MuiButton-outlined-74 MuiButton-sizeSmall-83 SceneCards-cardActionButton-3\\" role=\\"button\\" href=\\"/experience/vreditor/test/test\\"><span class=\\"MuiButton-label-71\\"><svg class=\\"MuiSvgIcon-root-89\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><g><path d=\\"M19 4H5c-1.11 0-2 .9-2 2v12c0 1.1.89 2 2 2h4v-2H5V8h14v10h-4v2h4c1.1 0 2-.9 2-2V6c0-1.1-.89-2-2-2zm-7 6l-4 4h3v6h2v-6h3l-4-4z\\"></path></g></svg></span><span class=\\"MuiTouchRipple-root-95\\"></span></a><a tabindex=\\"0\\" class=\\"MuiButtonBase-root-86 MuiButton-root-70 MuiButton-outlined-74 MuiButton-sizeSmall-83 SceneCards-cardActionButton-3\\" role=\\"button\\" href=\\"/experience/vreditor/test/test/mode-scene-edit\\"><span class=\\"MuiButton-label-71\\"><svg class=\\"MuiSvgIcon-root-89\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><g><path d=\\"M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z\\"></path></g></svg></span><span class=\\"MuiTouchRipple-root-95\\"></span></a></div></div></div>"`;
